lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -2024,6 +2025,10 @@ static void replaceUsesOfBuiltinVar(Value *V, const APInt &AccumulatedOffset,
       } else {
         // The function has an index parameter.
         if (auto *VecTy = dyn_cast<FixedVectorType>(Load->getType())) {
+          // Reconstruct the original global variable vector because
+          // the load type may not match.
+          // global <3 x i64>, load <6 x i32>
+          VecTy = cast<FixedVectorType>(GV->getValueType());
```

Comment:
I would also add in an `llvm_unreachable` call here if the Load and GV types aren't the same size.

Reply:
Added in latest commit, thanks!